app/models/user.rb

```diff
@@ -33,4 +33,8 @@ def destroy
   def to_s
     email
   end
+
+  def display_name
+    self[:display_name] || email
```

Comment:
I think we want this to be `self.user_key` - we're not always guaranteed to have an email if we integrate with a campus single sign on system.  Some use e-mail, some use a userID.  If I'm remembering right, devise aliases whatever it uses to synchronize with SSO ids as "user_key" or something similar.

Reply:
```
irb(main):001:0> u = User.first
  User Load (2.2ms)  SELECT  "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT $1  [["LIMIT", 1]]
=> #<User id: 1, email: "mark@curationexperts.com", created_at: "2021-12-04 23:31:23", updated_at: "2021-12-04 23:32:0...
irb(main):002:0> u.user_key
=> "mark@curationexperts.com"
irb(main):003:0> 
```

Reply:
In ETDs, for contrast, we use the PPID (Public Personal ID) to match users with shibboleth entities - you can't log in with a PPID, but because it's public, all systems can use it to uniquely identify users.  When you login, shibboleth gives you back the PPID, e-mail, and display name.  (But less fully configured systems might not give us display_name and e-mail, so but user_key is always safe to use)
```
irb(main):010:0> u = User.where(uid: 'mbusse2').first
=> #<User id: 31, email: "mbusse2@emory.edu", created_at: "2017-08-24 13:56:29", updated_at: "2022-01-13 18:57:03", gu...
irb(main):011:0> u.user_key
=> "P3753210"
irb(main):012:0> u.ppid
=> "P3753210"
irb(main):013:0> u.display_name
=> "Mark Bussey"
```